src/care/forall.h

```diff
@@ -71,14 +71,33 @@ namespace care {
       const int length = end - start;
 
       if (length != 0) {
+
 #ifndef CARE_DISABLE_RAJAPLUGIN
          RAJAPlugin::pre_forall_hook(ExecutionPolicyToSpace<ExecutionPolicy>::value, fileName, lineNumber);
 #endif
+
+#if CARE_ENABLE_PARALLEL_LOOP_BACKWARDS
+         RAJA::RangeStrideSegment rangeSegment(start, end, 1);
```

Comment:
No default constructor, eh?

Would it be better either define segmentStart, segmentEnd, and segmentStride as variables can call rangeSegment(segmentStart, segmentEnd, segmentStride)?  Or defined a boolean isParallel and initialize using a ternary operator?

Reply:
I'll go with the ternary operator.